packages/arb-token-bridge-ui/src/components/common/HeaderConnectWalletButton.tsx

```diff
@@ -1,27 +1,90 @@
-import { useWallet } from '@arbitrum/use-wallet'
 import { PlusCircleIcon } from '@heroicons/react/outline'
+import { ConnectButton } from '@rainbow-me/rainbowkit'
+import { HeaderAccountPopover } from './HeaderAccountPopover'
+import { HeaderNetworkInformation } from './HeaderNetworkInformation'
+import { NetworkSelectionContainer } from './NetworkSelectionContainer'
 
-import { modalProviderOpts } from '../../util/modelProviderOpts'
-
-export function HeaderConnectWalletButton() {
-  const { connect } = useWallet()
+function NetworkButtonAndConnectButton({
+  connected,
+  chainUnsupported,
+  openConnectModal,
+  openChainModal
+}: {
+  connected: boolean
+  chainUnsupported?: boolean
+  openChainModal: () => void
+  openConnectModal: () => void
+}) {
+  if (!connected) {
+    return (
+      <button
+        onClick={openConnectModal}
+        type="button"
+        className="arb-hover flex w-full flex-row items-center bg-lime-dark px-6 py-3 text-2xl font-medium text-white lg:rounded-full lg:bg-lime-dark lg:px-4 lg:py-2 lg:text-base lg:font-normal"
+      >
+        <PlusCircleIcon className="mr-3 h-10 w-10 rounded-full bg-white stroke-lime-dark p-1 opacity-40" />
+        Connect Wallet
+      </button>
+    )
+  }
 
-  async function showConnectionModal() {
-    try {
-      await connect(modalProviderOpts)
-    } catch (error) {
-      // Dialog was closed by user
-    }
+  if (chainUnsupported) {
+    return (
+      <button
+        onClick={openChainModal}
+        type="button"
+        className="arb-hover flex w-full flex-row items-center bg-[#FF494A] bg-lime-dark px-6 py-3 text-2xl font-medium text-white lg:rounded-full lg:px-4 lg:py-2 lg:text-base lg:font-normal"
+      >
+        Wrong network
+      </button>
+    )
   }
 
   return (
-    <button
-      onClick={showConnectionModal}
-      type="button"
-      className="arb-hover flex w-full flex-row items-center bg-lime-dark px-6 py-3 text-2xl font-medium text-white lg:rounded-full lg:bg-lime-dark lg:px-4 lg:py-2 lg:text-base lg:font-normal"
+    <div className="flex gap-3">
+      <NetworkSelectionContainer>
+        <HeaderNetworkInformation />
+      </NetworkSelectionContainer>
+      <HeaderAccountPopover />
+    </div>
+  )
+}
+
+function HeaderConnectWalletButtonWrapper({
+  children,
+  ready
+}: {
+  children: JSX.Element
+  ready: boolean
+}) {
+  return (
+    <div
+      className={!ready ? 'pointer-events-none select-none opacity-0' : ''}
+      aria-hidden={!ready}
     >
-      <PlusCircleIcon className="mr-3 h-10 w-10 rounded-full bg-white stroke-lime-dark p-1 opacity-40" />
-      Connect Wallet
-    </button>
+      {children}
+    </div>
+  )
+}
+
+export function HeaderConnectWalletButton() {
+  return (
+    <ConnectButton.Custom>
+      {({ account, chain, openChainModal, openConnectModal, mounted }) => {
+        const ready = mounted
```

Comment:
Why create another variable, could just make it `<HeaderConnectWalletButtonWrapper ready={mounted}>`?